sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -126,6 +128,9 @@ case class BroadcastExchangeExec(
             val relation = mode.transform(input, Some(numRows))
 
             val dataSize = relation match {
+              case EmptyHashedRelation =>
+                knownRowCount = Some(0L)
```

Comment:
See L118. We already know the row count and we even use it to check `numRows >= MAX_BROADCAST_TABLE_ROWS`